src/js/utils/portal.js

```diff
@@ -10,24 +11,32 @@ export const expectPortal = portalId => ({
   toMatchSnapshot: () => {
     const node = document.getElementById(portalId);
     if (node) {
-      const styles = css.parse(
-        document.getElementsByTagName('style')[0].innerHTML,
-      );
-      styles.stylesheet.rules = styles.stylesheet.rules.filter(rule => {
+      const styles = css
+        .parse(document.getElementsByTagName('style')[0].innerHTML)
+        .root();
+      styles.each(rule => {
         // skip everything that is not media or rule
-        if (['media', 'rule'].indexOf(rule.type) < 0) {
-          return false;
+        if (
+          ['atrule', 'rule'].indexOf(rule.type) < 0 ||
+          (rule.type === 'atrule' && rule.name !== 'media')
+        ) {
+          rule.remove();
         }
-        if (!rule.selectors) {
-          return true;
+        if (rule.selectors) {
+          const selector = rule.selectors.join('');
+          if (
+            !Array.from(node.classList).some(
+              className => selector.indexOf(className) >= 0,
+            )
+          ) {
+            rule.remove();
+          }
```

Comment:
Can you explain what is the motivation for these changes? Is it related to the package changes?

Reply:
Besides the errors, the `css` package does not seem to be maintained any longer and it is not a standalone package so its dependencies are very stale. `postcss` seems to be actively maintained. And so, the motivation, is to migrate Grommet to devDependencies that are maintained with security and bug fixes. `postcss` also greatly expands the capabilities of what can be done given its ecosystem of plugins (though not relevant for this discussion).

This snippet is just reproducing the same behavior of traversing the AST once the CSS is parsed by the tool. Because the specifications for the ASTs are different between `css` and `postcss` the logic needed to be slightly changed. The huge advantage that `postcss` gives for this purpose, in my opinion, is provide OOP-like functions to manipulate the AST instead of interacting directly with its structure.